File: Java/Stack/Stack.java
package stack;

public class Stack {

    Node top;

    Stack() {
        top = null;
    }

    boolean isEmpty() {
        return top == null;
    }

    void push(Node node) {
        if (isEmpty()) {
            top = node;
        } else {
            node.down = top;
            top = node;
        }
    }

    Node pop() {
        if (isEmpty()) {
            System.out.println("Stack is empty !");
            return null;
        } else {
            Node deleted = top;
            top = top.down;
            return deleted;
        }
    }

    void show() {
        Node temp = top;

        while (temp != null) {
            System.out.println(temp.number);
            temp = temp.down;
        }
    }
}

File: C-plus-plus/Stack/stack.cpp
//
// Created by berkan on 1.09.2021.
//

#include "stack.h"
#include <iostream>

using namespace std;

Stack::Stack()
{
    Stack::top = nullptr;
}

bool Stack::isEmpty()
{
    if (Stack::top == nullptr) return true;
    return false;
}

void Stack::add(Node *node)
{
    if (isEmpty())
    {
        Stack::top = node;
    } else
    {
        node->down = Stack::top;
        Stack::top = node;
    }
}

int Stack::del()
{
    if (isEmpty())
    {
        cout << "Stack is empty" << endl;
    } else
    {
        Node * tmp = Stack::top;
        int deleted = tmp->number;
        Stack::top = Stack::top->down;
        delete(tmp);
        return deleted;
    }
}

void Stack::show()
{
    Node * tmp = Stack::top;

    while (tmp != nullptr)
    {
        cout << tmp->number << endl;
        tmp = tmp->down;
    }
}